### PR TITLE
feat(@angular/cli): add cache build option

### LINKED ISSUE #7467: differential builds without watch mode
### Bug Report or Feature Request (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
```

### Versions.

```
@angular/cli: 1.3.0 (e)
node: 6.10.2
os: darwin x64
@angular/common: 4.3.5
@angular/compiler: 4.3.5
@angular/core: 4.3.5
@angular/forms: 4.3.5
@angular/http: 4.3.5
@angular/platform-browser: 4.3.5
@angular/platform-browser-dynamic: 4.3.5
@angular/router: 4.3.5
@angular/cli: 1.3.0
@angular/compiler-cli: 4.3.5
```

### Desired functionality.

I'd like `ng build` to be just as fast on rebuilds as `ng watch` is.
In other words, is it possible to tell `ng build` to keep (on disk) the same intermediate data that `ng watch` does to make recompiles fast?

### Mention any other details that might be useful.

We  use a `GNUmakefile`, and I've plumbed `ng build` into that makefile. For most of my team members, it's easier just to tell them to type `make` (which also builds our server-side) rather than teach them how to use the `ng`tools and to ask them to have a watch process running in the background.

### PR DESCRIPTION
**Pull request by filipesilva**
This is an experimental option to use a cache between builds.

Fix #7467

### CODE REVIEW

**Comment by filipesilva** [comment]
Blocked by:

- [ ]: https://github.com/mzgoddard/hard-source-webpack-plugin/pull/202

(this list will increase as new problems are found)

**Comment by intellix** [comment]
Perhaps a fix/improvement for multiple i18n builds? :) https://github.com/angular/angular-cli/issues/6789

**Comment by filipesilva** [comment]
Also want to benchmark https://github.com/webpack-contrib/cache-loader gains here.

File: packages/@angular/cli/commands/build.ts [hunk 1]
```diff
@@ -207,6 +207,13 @@ export const baseBuildCommandOptions: any = [
     type: Boolean,
     description: 'Flag to prevent building an app shell',
     default: false
+  },
+  {
+    name: 'cache',
+    type: Boolean,
+    description: '(Experimenta) Use cache between builds.',
```
Comment (JulienBourgain) [inline packages/@angular/cli/commands/build.ts]: Experimental with a final "L"

**Comment by Fank** [comment]
@filipesilva the linked webpack issue has been merged.
any progress on this one?

**Comment by filipesilva** [comment]
I experimented on this a bit more in https://github.com/angular/angular-cli/pull/9295, and we merged that one instead.

**Pull request closed without merging**